server/middlewares/article.middlewares.js

```diff
@@ -97,11 +97,99 @@ const validateHighlight = (req, res, next) => {
   return next();
 };
 
+const reviewArticleValidator = (req, res, next) => {
+  const { articleId } = req.params;
+  if (!Object.keys(req.body).length) {
+    return res.status(400).json({
+      errors: {
+        body: ['No input provided'],
+      },
+    });
+  }
+
+  if (!validations.verifyUUID(articleId)) {
+    return res.status(400).json({
+      errors: {
+        body: ['id not valid'],
+      },
+    });
+  }
+
+  if (!req.user.userObj.isReviewer) {
+    return res.status(403).json({
+      errors: {
+        body: ['User is not a reviewer'],
+      },
+    });
+  }
+
+  if (!req.body.reviewer_comment) {
+    return res.status(400).json({
+      errors: {
+        body: ['Reviewer comment is required'],
+      },
+    });
+  }
+
+  return next();
+};
+
+const articleStatusValidator = (req, res, next) => {
+  const { articleId } = req.params;
+  if (!Object.keys(req.body).length) {
+    return res.status(400).json({
+      errors: {
+        body: ['No input provided'],
+      },
+    });
+  }
+
+  if (!validations.verifyUUID(articleId)) {
+    return res.status(400).json({
+      errors: {
+        body: ['id not valid'],
+      },
+    });
+  }
+
+  if (!req.user.userObj.isAdmin) {
+    return res.status(401).json({
+      errors: {
+        body: ['User is not an admin'],
+      },
+    });
+  }
+
+  const { status: adminStatus } = req.body;
+
+  if (!adminStatus) {
+    return res.status(400).json({
+      errors: {
+        body: ['Status is required'],
+      },
+    });
+  }
+
+  const status = adminStatus.toLowerCase();
+  if (!(status === 'accepted' || status === 'rejected')) {
+    return res.status(400).json({
+      errors: {
+        body: ['Status can only be accepted or rejected'],
+      },
+    });
+  }
```

Comment:
I saw a similar check like this inside the `article.controller` lines 309-311 any reason we are doing this twice? @sojida

Reply:
Yes. In the middleware, we are checking the `req.body.status` to accept only `accepted` and `rejected`. In the controller, after we have found the reported article from the DB, we are checking if the reported article has already been given a status of `accepted` or `rejected`. We want the admin to make the action once.